hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/MirrorEntityAccess.java

```diff
@@ -0,0 +1,106 @@
+ * ‍
+ */
+
+import static com.hedera.mirror.common.domain.entity.EntityType.TOKEN;
+import static com.hedera.services.evm.accounts.HederaEvmContractAliases.isMirror;
+
+import com.google.protobuf.ByteString;
+import java.util.Optional;
+import javax.inject.Named;
+import lombok.RequiredArgsConstructor;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.units.bigints.UInt256;
+import org.hyperledger.besu.datatypes.Address;
+
+import com.hedera.mirror.common.domain.entity.Entity;
+import com.hedera.mirror.common.domain.entity.EntityIdEndec;
+import com.hedera.mirror.common.util.DomainUtils;
+import com.hedera.mirror.web3.repository.ContractRepository;
+import com.hedera.mirror.web3.repository.ContractStateRepository;
+import com.hedera.mirror.web3.repository.EntityAccessRepository;
+import com.hedera.services.evm.store.contracts.HederaEvmEntityAccess;
+
+@Named
+@RequiredArgsConstructor
+public class MirrorEntityAccess implements HederaEvmEntityAccess {
+    private final EntityAccessRepository entityRepository;
+    private final ContractRepository contractRepository;
+    private final ContractStateRepository contractStateRepository;
+
+    @Override
+    public boolean isUsable(Address address) {
+        //TODO impl
+        return false;
+    }
+
+    @Override
+    public long getBalance(Address address) {
+        final var entity = findEntity(address);
+        return entity.isPresent() ? entity.get().getBalance() : 0L;
+    }
+
+    @Override
+    public boolean isExtant(Address address) {
+        return entityRepository.existsById(entityIdFromEvmAddress(address));
+    }
+
+    @Override
+    public boolean isTokenAccount(Address address) {
+        final var entity = findEntity(address);
+        return entity.isPresent() && entity.get().getType().equals(TOKEN);
+    }
+
+    @Override
+    public ByteString alias(Address address) {
+        final var entity = findEntity(address);
+        return entity.map(value -> ByteString.copyFrom(value.getAlias())).orElse(ByteString.EMPTY);
+    }
+
+    @Override
+    public UInt256 getStorage(Address address, Bytes key) {
+        final var storage = contractStateRepository.findStorage(entityIdFromEvmAddress(address),
+                key.toArrayUnsafe());
+        final var storageBytes = storage.map(Bytes::wrap).orElse(Bytes.EMPTY);
+        return UInt256.fromBytes(storageBytes);
+    }
+
+    @Override
+    public Bytes fetchCodeIfPresent(Address address) {
+        final var runtimeCode = contractRepository.findContractCode(entityIdFromEvmAddress(address));
+        return runtimeCode.map(Bytes::wrap).orElse(Bytes.EMPTY);
+    }
+
+    private Optional<Entity> findEntity(Address address) {
+        if (isMirror(address.toArrayUnsafe())) {
+            final var entityId = entityIdFromEvmAddress(address);
+            return entityRepository.findByIdAndDeletedFalse(entityId);
+        } else {
+            //TODO queryBy alias field instead
+            return null;
```

Comment:
Can copy from `EntityRepository.findByEvmAddress` in importer. 
```suggestion
            return entityRepository.findByEvmAddress(addressBytes);
```

Reply:
Hey Steven, thanks for the detailed comments and suggestions so far. The received address field contains a bytes representation of the account alias.And since the `alias` field in the entity table is also indexed, I introduced a method `findByAliasAndDeletedFalse` querying the entity by it  instead of the `evmAddress` to avoid additional parsing.

Reply:
Ok, was not aware that address can also be an alias. It's weird that it can be a 20 byte EVM address (shard/realm/num format) but not a regular 20 byte EVM address. Are you sure it can't be all 3 and we should have a findByEvmAddress lookup as well?

Reply:
Actually, for contract creates there might be evm_address fields with CREATE1 (keccak256 calculated address using nonces) or CREATE2 addreses, both 20-bytes, so we should be also able to search results by evm_address. We will take into account also these cases.